---
title: "Phylogenetics_SRV_viruses"
author: "Anton Zhelonkin"
date: "5/24/2022"
output: html_document
---

## Specifications
```{r}
version
```

## Requirements
```{r}
if (!("reutils" %in% installed.packages()))
install.packages("reutils")
if (!("XML" %in% installed.packages()))
install.packages("XML")
if (!("RCurl" %in% installed.packages()))
install.packages("RCurl")

library(reutils)
```

```{r}
sessionInfo()
```


# Description 
This is a replication analysis of a paper by Zao et al. *"A novel simian retrovirus subtype discovered in cynomolgus monkeys (Macaca fascicularis)"*. The plan is simple: 
* download sequences by assesion numbers from GenBank (db-nucleotide)
* align sequences of the newly discovered viral strain SRV 
* draw a phylogenetic tree comparing the new strain with the previously known strains 

# 1. Data preparation
### Preparing for data extraction 
```{r}
options(reutils.email = "good.tony@yandex.ru") # i`m a real person
```

## find UIDs
```{r}
# newly discovered SRV viral whole genomes search by assession number given in the article 
srv_new1 <- esearch(db = "nucleotide", term = "KU605777")
srv_new2 <- esearch(db = "nucleotide", term = "KU605778")
srv_new3 <- esearch(db = "nucleotide", term = "KU605779")
srv_new4 <- esearch(db = "nucleotide", term = "KU605780")

# previously known SRV viruses 
srv1 <- esearch(db = "nucleotide", term = "M11841")
srv2 <- esearch(db = "nucleotide", term = "AF126467")
srv3 <- esearch(db = "nucleotide", term = "M12349")
srv4 <- esearch(db = "nucleotide", term = "FJ971077")
srv5 <- esearch(db = "nucleotide", term = "AB611707")
srv6_env <- esearch(db = "nucleotide", term = "AY598468")
srv7_pol <- esearch(db = "nucleotide", term = "AY594212")
serv <- esearch(db = "nucleotide", term = "U85505")

# all UIDs vector
uid <- c(srv_new1[1], srv_new2[1], srv_new3[1], srv_new4[1],
         srv1[1], srv2[1], srv3[1], srv4[1], srv5[1], srv6_env[1], srv7_pol[1], serv[1])
```

## extract FASTA sequences by uids
```{r}
srv_new <- efetch(uid[1:4], db = "nucleotide", rettype = "fasta", retmode = "text")

srv_seq_f <- efetch(uid[1:12], db = "nucleotide", rettype = "fasta", retmode = "text")
```

## write FASTA file with new SRV genomes and all known SRV viruses with the newly discovered ones 
```{r}
write(content(srv_new), "srv_new_sequences")

write(content(srv_seq_f), "srv_all_sequences")
```



# Drawing tree based on modeltest-raxml pipeline

```{r}
if (!("ggtree" %in% installed.packages()))
BiocManager::install("ggtree")

library(ggtree)
```


```{r}
srv_tr <- read.tree("srv_all_raxml.raxml.bestTree")
ggtree(srv_tr) + geom_tiplab() + xlim(0,0.8)
```
sabing tree
```{r}
png("srv_all_raxml_bestTree.png")
ggtree(srv_tr) + geom_tiplab() + xlim(0,0.8)
dev.off()
```


### Renaming tip labels in tree
```{r}
srv_tr_ren <- srv_tr
srv_tr_ren$tip.label <- c("SRV-5", "SRV-4", "SRV8/TEX/2015/V2", "SRV8/TEX/2015/V1",
                          "SRV8/TEX/2015/V3", "SRV8/SUZ/2012", "SRV-3", "SRV-1", "SRV-7",
                          "SERV", "SRV-6", "SRV-2")
ggtree(srv_tr_ren, ladderize=TRUE)+ geom_tiplab() + xlim(0,0.8) + geom_treescale(width = 0.05)
```
saving tree
```{r}
png("srv_all_raxml_named_bestTree.png")
ggtree(srv_tr_ren, ladderize=TRUE)+ geom_tiplab() + xlim(0,0.8) + geom_treescale(width = 0.05)
dev.off()
```

# Drawing tree based on PhyMl3 Bootstrap 100 
Analysis run at http://phylogeny.lirmm.fr/phylo_cgi/one_task.cgi?task_type=phyml
* Substitution model: GTR  
* Bootstrapped data sets: 100

```{r}
phy_tree <- read.tree("phyml3_all_tree.nwk")
ggtree(phy_tree) + geom_tiplab() + xlim(0,2)
```
### Renaming tip labels in tree
```{r}
phy_tree_ren <- phy_tree
phy_tree_ren$tip.label <- c("SRV-4", "SRV8/SUZ/2012", "SRV8/TEX/2015/V1",
                            "SRV8/TEX/2015/V2", "SRV8/TEX/2015/V3", "SRV-5",
                            "SRV-3", "SRV-1", "SRV-2", "SRV-6 env", "SERV",
                            "SRV-7 pol")
ggtree(phy_tree_ren, ladderize=TRUE, scale = 0.5)+ geom_tiplab() + xlim(0,2) + geom_treescale()
```
```{r}
ggtree(phy_tree_ren, ladderize=TRUE)+ geom_tiplab() + xlim(0,1.8) + 
  geom_treescale(width = 0.05)
```
saving tree
```{r}
png("phyml3_all_tree.png")
ggtree(phy_tree_ren, ladderize=TRUE)+ geom_tiplab() + xlim(0,1.8) + 
  geom_treescale(width = 0.05)
dev.off()
```

